cpp/oneapi/dal/algo/knn/common.hpp

```diff
@@ -69,6 +79,34 @@ using v1::by_default;
 
 } // namespace method
 
+namespace optional_results {
+
+class namespace_id {};
+using optional_result_id_t = optional_result_id<namespace_id>;
+
+} // namespace optional_results
```

Comment:
Suggest to optional results id type defined in the algorithm's namespace, e.g., `oneapi::dal::knn`